### PR TITLE
Feat: More options for deno file upload

### PR DESCRIPTION
**Pull request by Meldiron**
Test code:

```ts
import * as sdk from "./mod.ts";

const client = new sdk.Client();

client
    .setEndpoint('https://8080-appwrite-appwrite-m65a55s4hpc.ws-eu47.gitpod.io/v1') // Your API Endpoint
    .setProject('dev') // Your project ID
    .setKey('aae5cee0401ddc69d8f80f3267e817f2bec341548a4ff06aab9e766b356455aecb49f6190f08de6c7d4bebfd8f145f20f78a5257a5b9b3bb9d9477e56bb6e72c18c088e5638ef437b7d4488e0ad391e2f7a91b541898eaea2816541c1ed832cf41b31e0db3e685666b3d9b78e06de80c2b093b9f8386d327acd63665e80c59db') // Your secret API key
;

let storage = new sdk.Storage(client);

let promise = storage.createFile('b1', 'unique()', sdk.InputFile.fromPath("./vid.mp4", "vid.mp4"), undefined, undefined, (ch) => {
    console.log(ch);
});

promise.then(function (response) {
    console.log(response);
}, function (error) {
    console.log(error);
});
```

Output:
```
gitpod /workspace/sdk-generator/examples/deno (feat-improve-deno-file-upload) $ deno run -A app.ts 
{
  "$id": "62b59933efeb57e020e6",
  progress: 29.388596145314043,
  sizeUploaded: 5242880,
  chunksTotal: 4,
  chunksUploaded: 1
}
{
  "$id": "62b59933efeb57e020e6",
  progress: 58.777192290628086,
  sizeUploaded: 10485760,
  chunksTotal: 4,
  chunksUploaded: 2
}
{
  "$id": "62b59933efeb57e020e6",
  progress: 88.16578843594213,
  sizeUploaded: 15728640,
  chunksTotal: 4,
  chunksUploaded: 3
}
{
  "$id": "62b59933efeb57e020e6",
  progress: 100,
  sizeUploaded: 17839846,
  chunksTotal: 4,
  chunksUploaded: 4
}
{
  "$id": "62b59933efeb57e020e6",
  bucketId: "b1",
  name: "vid.mp4",
  dateCreated: 1656068403,
  signature: "447b0002cad41f61551e7bf79ef6c7f9",
  mimeType: "video/mp4",
  sizeOriginal: 17839845,
  chunksTotal: 4,
  chunksUploaded: 4,
  "$permissions": { read: [], write: [] }
}
```

- [x] Video has format
- [x] Video has name
- [x] Video can be played (all 30s)

Generated function after changes:

```ts
    /**
     * Create File
     *
     * Create a new file. Before using this route, you should create a new bucket
     * resource using either a [server
     * integration](/docs/server/database#storageCreateBucket) API or directly
     * from your Appwrite console.
     * 
     * Larger files should be uploaded using multiple requests with the
     * [content-range](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Range)
     * header to send a partial request with a maximum supported chunk of `5MB`.
     * The `content-range` header values should always be in bytes.
     * 
     * When the first request is sent, the server will return the **File** object,
     * and the subsequent part request must include the file's **id** in
     * `x-appwrite-id` header to allow the server to know that the partial upload
     * is for the existing file and not for a new one.
     * 
     * If you're creating a new file using one of the Appwrite SDKs, all the
     * chunking logic will be managed by the SDK internally.
     * 
     *
     * @param {string} bucketId
     * @param {string} fileId
     * @param {InputFile} file
     * @param {string[]} read
     * @param {string[]} write
     * @throws {AppwriteException}
     * @returns {Promise}
     */
    async createFile(bucketId: string, fileId: string, file: InputFile, read?: string[], write?: string[], onProgress = (progress: UploadProgress) => {}): Promise<Models.File> {
        if (typeof bucketId === 'undefined') {
            throw new AppwriteException('Missing required parameter: "bucketId"');
        }

        if (typeof fileId === 'undefined') {
            throw new AppwriteException('Missing required parameter: "fileId"');
        }

        if (typeof file === 'undefined') {
            throw new AppwriteException('Missing required parameter: "file"');
        }

        let path = '/storage/buckets/{bucketId}/files'.replace('{bucketId}', bucketId);
        let payload: Payload = {};

        if (typeof fileId !== 'undefined') {
            payload['fileId'] = fileId;
        }
        if (typeof file !== 'undefined') {
            payload['file'] = file;
        }
        if (typeof read !== 'undefined') {
            payload['read'] = read;
        }
        if (typeof write !== 'undefined') {
            payload['write'] = write;
        }

        const size = file.size;
        
        const headers: { [header: string]: string } = {
            'content-type': 'multipart/form-data',
        };

        let id: string | undefined = undefined;
        let response: any = undefined;

        let chunksUploaded = 0;

        if(fileId != 'unique()') {
            try {
                response = await this.client.call('get', path + '/' + fileId, headers);
                chunksUploaded = response.chunksUploaded;
            } catch(e) {
            }
        }

        let currentChunk = 1;
        let currentPosition = 0;
        let uploadableChunk = new Uint8Array(Client.CHUNK_SIZE);

        const uploadChunk = async (lastUpload = false) => {
            if(currentChunk <= chunksUploaded) {
                return;
            }

            const start = ((currentChunk - 1) * Client.CHUNK_SIZE);
            const end = start + currentPosition;

            if(!lastUpload || currentChunk !== 1) {
                headers['content-range'] = 'bytes ' + start + '-' + end + '/' + size;
            }

            let uploadableChunkTrimmed: Uint8Array;
            
            if(currentPosition + 1 >= Client.CHUNK_SIZE) {
                uploadableChunkTrimmed = uploadableChunk;
            } else {
                uploadableChunkTrimmed = new Uint8Array(currentPosition + 1);
                for(let i = 0; i <= currentPosition; i++) {
                    uploadableChunkTrimmed[i] = uploadableChunk[i];
                }
            }

            if (id) {
                headers['x-appwrite-id'] = id;
            }

            payload['file'] = { type: 'file', file: new File([uploadableChunkTrimmed], file.name), filename: file.name };

            response = await this.client.call('post', path, headers, payload);

            if (!id) {
                id = response['$id'];
            }

            if (onProgress !== null) {
                onProgress({
                    $id: response['$id'],
                    progress: Math.min((currentChunk) * Client.CHUNK_SIZE, size) / size * 100,
                    sizeUploaded: end+1,
                    chunksTotal: response['chunksTotal'],
                    chunksUploaded: response['chunksUploaded']
                });
            }

            uploadableChunk = new Uint8Array(Client.CHUNK_SIZE);
            currentPosition = 0;
            currentChunk++;
        }

        for await (const chunk of file.stream) {
            let i = 0;
            for(const b of chunk) {
                uploadableChunk[currentPosition] = chunk[i];

                if(currentPosition + 1 >= Client.CHUNK_SIZE) {
                    await uploadChunk();
                    currentPosition--;
                }

                i++;
                currentPosition++;
            }
        }

        await uploadChunk(true);

        return response;
    }
```

### CODE REVIEW

**Comment by TorstenDittmann** [comment]
Let's get some second 👀 on this as well

File: templates/deno/src/services/service.ts.twig [hunk 1]
```diff
@@ -70,122 +71,119 @@ export class {{ service.name | caseUcfirst }} extends Service {
...
-                        sizeUploaded: end+1,
-                        chunksTotal: response['chunksTotal'],
-                        chunksUploaded: response['chunksUploaded']
-                    });
-                }
+            if (id) {
+                headers['x-{{spec.title | caseLower }}-id'] = id;
+            }
+
+            payload['{{ parameter.name }}'] = { type: 'file', file: new File([uploadableChunkTrimmed], {{ parameter.name | caseCamel | escapeKeyword }}.name), filename: {{ parameter.name | caseCamel | escapeKeyword }}.name };
```
Comment (lohanidamodar) [inline templates/deno/src/services/service.ts.twig:146]:
```suggestion
            payload['{{ parameter.name }}'] = { type: 'file', file: new File([uploadableChunkTrimmed], {{ parameter.name | caseCamel | escapeKeyword }}.name), filename: {{ parameter.name | caseCamel | escapeKeyword }}.filename };
```

**Review by lohanidamodar — CHANGES_REQUESTED** [review]
Looks good. Small change and also may be we should add test for buffer, stream and plaintext?

**Comment by lohanidamodar** [comment]
@Meldiron Also needs sync

File: templates/deno/src/services/service.ts.twig (hunk 1 above)
Comment (Meldiron) [reply templates/deno/src/services/service.ts.twig:146]: Renamed to filename

**Pull request merged**